src/Mod/AddonManager/addonmanager_workers.py

```diff
@@ -355,24 +360,50 @@ def run(self):
             url = self.repos[self.idx][1]
             self.info_label.emit(translate("AddonsInstaller", "Retrieving info from") + ' ' + str(url))
             desc = ""
-            # get the README if possible
-            readmeurl = utils.getReadmeUrl(url)
-            if not readmeurl:
-                print("Debug: README not found for",url)
-            u = utils.urlopen(readmeurl)
-            if not u:
-                print("Debug: README not found at",readmeurl)
-            if u:
-                p = u.read()
-                if sys.version_info.major >= 3 and isinstance(p, bytes):
-                    p = p.decode("utf-8")
-                u.close()
-                readmeregex = utils.getReadmeRegex(url)
-                if readmeregex:
-                    readme = re.findall(readmeregex,p,flags=re.MULTILINE|re.DOTALL)
+            regex = utils.getReadmeRegex(url)
+            if regex:
+                # extract readme from html via regex
+                readmeurl = utils.getReadmeHTMLUrl(url)
+                if not readmeurl:
+                    print("Debug: README not found for",url)
+                u = utils.urlopen(readmeurl)
+                if not u:
+                    print("Debug: README not found at",readmeurl)
+                u = utils.urlopen(readmeurl)
+                if u:
+                    p = u.read()
+                    if sys.version_info.major >= 3 and isinstance(p, bytes):
+                        p = p.decode("utf-8")
+                    u.close()
+                    readme = re.findall(regex,p,flags=re.MULTILINE|re.DOTALL)
                     if readme:
-                        desc += readme[0]
-            if not desc:
+                        desc = readme[0]
+                else:
+                    print("Debug: README not found at",readmeurl)
+            else:
+                # convert raw markdown using lib
+                readmeurl = utils.getReadmeUrl(url)
+                if not readmeurl:
+                    print("Debug: README not found for",url)
+                u = utils.urlopen(readmeurl)
+                if u:
+                    p = u.read()
+                    if sys.version_info.major >= 3 and isinstance(p, bytes):
+                        p = p.decode("utf-8")
+                    u.close()
+                    desc = utils.fixRelativeLinks(p,readmeurl.rsplit("/README.md")[0])
+                    try:
+                        if NOMARKDOWN:
+                            raise ImportError
+                        import markdown  # try to use system Markdown lib
+                        desc = markdown.markdown(desc,extensions=['md_in_html'])
```

Comment:
So just to clarify, `markdown` is completely optional, right?

Reply:
> So just to clarify, markdown is completely optional, right?

yes

> By the way, I would test for markdown at the beginning of the file

Since python caches imports, this is more of a style choice. Personally, I prefer imports at the beginning of file, but in this module this is not the case.
Besides, here this is actually beneficial. Currently all addons are in github, so code would not hit this import and Markdown would not be loaded at all.

Reply:
Importing things in the middle of the code and not at the top is a bad practice that hides errors. We should avoid this. Just because it's used in the code doesn't mean it's a good practice, it's one of those things that we want to change, but it takes time.

Reply:
I could do a code cleanup for this module in second commit

Reply:
Will you add more stuff here @matslevsklysv ? Or is it OK for merge?